lib/web_ui/lib/src/engine/text/paragraph.dart

```diff
@@ -80,6 +82,18 @@ class EngineLineMetrics implements ui.LineMetrics {
   @override
   final double width;
 
+  /// The full width of the line including all trailing space but not new lines.
+  ///
+  /// The difference between [width] and [widthWithSpaces] is that
+  /// [widthWithSpaces] includes trailing spaces in the width calculation
+  /// while [width] doesn't.
+  ///
+  /// For alignment purposes for example, the [width] property is the right one
+  /// to use because trailing spaces shouldn't affect the centering of text.
+  /// But for placing cursors in text fields, we do care about trailing
+  /// spaces so [widthWithSpaces] is more suitable.
+  final double widthWithSpaces;
```

Comment:
Does it have any effect on leading spaces? If not, maybe call this `widthWithTrailingSpaces`?

Reply:
Good call.